Library/MapleBacon/MapleBacon/MapleBaconConfig.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  MapleBaconConfig.swift
+//  MapleBacon
+//
+//  Created by Danilo Topalovic on 10.03.16.
+//  Copyright © 2016 Zalando SE. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+/**
+ * MapleBacon Config
+ */
+public class MapleBaconConfig {
+    
+    /// singleton param
+    public static let sharedConfig: MapleBaconConfig = MapleBaconConfig()
+    
+    /// Holds the storage part of the config
+    var storage: Storage
+    
+    /**
+     Private Constructor
+     inits default config
+     */
+    private init() {
+        self.storage = MapleBaconConfig.Storage.defaultStorage()
+    }
+    
+    /**
+     * Storage Class
+     * Holds all configuration for the storage
+     * part of MapleBacon
+     */
+    class Storage {
+        
+        /// used for idents and/or paths of the storage engines
+        var defaultStorageName: String = ""
+        /// used for dispatch and/or serial queues
+        var queueLabel: String = ""
+        /// basepath of the storage (disk)
+        var baseStoragePath: String = ""
+        /// the rootUUIDNamespace should be either ns:URL or ns:DNS -> http://www.ietf.org/rfc/rfc4122.txt
+        var rootUUIDNamespace: String = ""
+        /// should be the bundle-id or other unique app ident
+        var baseUUIDName: String = ""
+        /// use UUID Idents instead of clean sha1
```

Comment:
Most of what you said is true. I've added them to make sure everything goes properly in the readme at the end. If you think about adding MapleBacon to your project with cocoapods this file is the only actual touchpoint for you. I will remove this once I work on the readme